metrics/internal/collectors/ceph-cluster_test.go

```diff
@@ -217,5 +217,88 @@ func TestCollectMirrorinDaemonCount(t *testing.T) {
 			}
 		}
 	}
+}
+
+func TestCollectLegacyOSDCount(t *testing.T) {
+	mockOpts.StopCh = make(chan struct{})
+	defer close(mockOpts.StopCh)
+
+	mockCollector := getMockCephClusterCollector(t, mockOpts)
+	mockCollector.AllowedNamespaces = append(mockCollector.AllowedNamespaces, "namespace1", "namespace2")
+	reason := "LVM-based OSDs on a PVC are deprecated, see documentation on replacing OSDs"
```

Comment:
Please add tests for Status with multiple/different reasons for completeness.

Reply:
ack, we want to drop them in the test?

Reply:
We want to ensure that we don't count other reasons.